### ISSUE TITLE
acount.createOAuth2Session return's null

### ISSUE THREAD

**Issue by TheKikeros** [appwrite/appwrite#1009]
i'm using version 0.7.0

```
`
class AppwriteAuthenticationRepoImpl extends AuthenticationRepository {
  Client _client;
  Account _account;
  AppwriteAuthenticationRepoImpl() {
    _client = Client();
    _account = Account(_client);
    _client.setEndpoint("https://kikero.net/v1").setProject("my project id");
  }

  @override
  Future<User> getCurrentUser() async {
    var result = await _account.get();
  }

  @override
  Future<Tuple<User, bool>> signInUser(
      BuildContext context, AuthProvider provider) async {
    var result = await _account.createOAuth2Session(
      provider: 'google',
    );
//*******************************result is null but the session is created successfully and when i do _account.get() i get the user *************************//
var userRaw = await _account.get();
    return Tuple(item1: User(email: "some Mail", name: "some name"));
  }

  @override
  Future<void> signOutUser() async {
    _account.deleteSession(sessionId: 'current');
  }
}
`
```

**Comment by wess** [comment]
Hitting this too, I would like to be able to use the response token to do things like post to Facebook/instagram. 

Thanks!

**Comment by Luuuuuis** [comment]
As I found out with the help of a nice team member (@christyjacob4), `createOAuth2Session` only redirects a user to the corresponding app. It cannot return any use, because it redirects to the consent screen which redirects back to the appwrite server. Then you can call `account.get()` to get their information.
So I would recommend to call nothing after `createOAuth2Session`.

Therefore it's actually not a bug. [This behavior is also documented.](https://appwrite.io/docs/client/account#accountCreateOAuth2Session)

**Comment by Luuuuuis** [comment]
> Hitting this too, I would like to be able to use the response token to do things like post to Facebook/instagram.
> 
> Thanks!

You can call in v0.8.x `account.getSessions()`. It does return the provider, providerUid aswell as the providerToken aka. access token. Unfortunately as of now it does not automatically refresh.

**Comment by eldadfux** [comment]
Appwrite 0.9 will introduce a new endpoint to fetch a single session

**Comment by Luuuuuis** [comment]
Are the Access tokens gonna be refreshed automatically? That would be a nice addition

**Comment by eldadfux** [comment]
This is now fixed and available as part of Appwrite 0.9:
https://dev.to/appwrite/announcing-appwrite-0-9-the-open-source-firebase-alternative-53ho 🥳 🎉
